src/mpi/datatype/type_get_name.c

```diff
@@ -55,10 +55,10 @@ static mpi_names_t mpi_names[] = {
     type_name_entry(MPI_FLOAT),
     type_name_entry(MPI_DOUBLE),
     type_name_entry(MPI_LONG_DOUBLE),
-    /* LONG_LONG_INT is allowed as an alias; we don't make it a separate
+    /* LONG_LONG is a synonym of LONG_LONG_INT; we don't make it a separate
      * type */
-/*    type_name_entry(MPI_LONG_LONG_INT), */
-    type_name_entry(MPI_LONG_LONG),
+/*    type_name_entry(MPI_LONG_LONG), */
+    type_name_entry(MPI_LONG_LONG_INT),
```

Comment:
After some more thought, I'm tempted to say drop this patch. I'm not sure there's cause enough to change 15 year old behavior. The test changes we made should be sufficient to resolve the issue we were seeing. What do you think?

Reply:
We can only change the tests for now, but I am not sure if it is a correct MPI behavior that returns LONG_LONG for LONG_LONG_INT. Maybe we can check with forum to figure out ?

Reply:
The tests needs to be further changed to cover `LONG_LONG_INT` with name `LONG_LONG`. I will delete the above patch.

Reply:
OK, sorry for the churn. We can certainly ask the forum what they think the correct behavior is in this case.

Reply:
I checked Open MPI and they return "MPI_LONG_LONG_INT" for both. Similarly, they return "MPI_C_COMPLEX" for MPI_C_FLOAT_COMPLEX. This matches up with my original interpretation of the standard. I think we should match what they do and return the canonical type name even for these synonym types.

Reply:
good catch for `MPI_C_FLOAT_COMPLEX`. I will fix it in this PR. So I will not discard the above name change commit.